### PR TITLE
Features/browser compat

### PR DESCRIPTION
**Pull request by ChronicStone**
- Removes dependency to node util by using extends
- Leverages captureStackTrace to fix stack trace issue
- Uses rollup for proper cjs/esm build
- Uses ESM locally
- Adds standalone ts declarations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,71 +1,71 @@
-# Kosa
+  # Kosa
```
Comment (satblip) [inline README.md]: I do not see the point of adding spacing in front of everything here. Or Am I missing something?

File: package.json [hunk 2]
```diff
@@ -40,6 +46,7 @@
     "chai": "4.0.0",
     "mocha": "7.2.0",
     "mocha-standard": "1.0.0",
+    "rollup": "^4.24.0",
```
Comment (satblip) [inline package.json]: We fixe the versions because not everyone is using semver the same way (unfortunately)

File: README.md (hunk 1 above)
Comment (satblip) [reply README.md]: Also would be nice to add a small section explaining how to use it in client-app (with the now inherit-ex that is a dependency)

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        meta?: ErrorMeta
+    );
+
+    statusCode: number;
+    message: string;
+    meta: ErrorMeta;
+    stack?: string;
+}
+
+export = Kosa;
```
Comment (satblip) [inline index.d.ts]: Needs an EOF

File: package.json [hunk 4]
```diff
@@ -40,7 +50,11 @@
...
     "semistandard": "11.0.0",
     "snazzy": "7.0.0"
+  },
+  "dependencies": {
+    "inherits-ex": "^2.1.0"
```
Comment (satblip) [inline package.json]: We fixe the versions because not everyone is using semver the same way (unfortunately)

File: test/ManagedError.spec.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
-const chai = require('chai');
-const Kosa = require('../src/index');
+import chai from "chai";
```
Comment (satblip) [inline test/ManagedError.spec.js]: we keep the single quote

File: test/ManagedError.spec.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
...
+import Kosa from "../src/index.js"
```
Comment (satblip) [inline test/ManagedError.spec.js]: we keep the single quote

File: README.md (hunk 1 above)
Comment (ChronicStone) [reply README.md]: I have no idea how nor why the modifications on the readme slipped in my commit, that's 100% not intentional. I'm going to be checking.

Regarding client-side usage, I don't see how it'd differ from server-side case ?

Comment (satblip) [reply README.md]: Mmh indeed, in a build process, the dependency tree will be flattened. I was thinking about a pure-html/js usage with then a `<script>` tag with the dep will be needed. But maybe not required to document; I think we will never use it in that scenario. 👍🏻

**Pull request merged**